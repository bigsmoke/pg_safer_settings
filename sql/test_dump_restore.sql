\o /dev/null
select not :{?test_stage} as test_stage_missing, not :{?extension_name} as extension_name_missing;
\o
\gset
\if :test_stage_missing
    \warn 'Missing `:test_stage` variable.'
    \quit
\endif
\if :extension_name_missing
    \warn 'Missing `:extension_name` variable.'
    \quit
\endif
\o /dev/null
select :'test_stage' = 'pre-dump' as in_pre_dump_stage;
\o
\gset

\set SHOW_CONTEXT 'errors'

\if :in_pre_dump_stage
    create extension pg_safer_settings with cascade;
\endif

call test_dump_restore__pg_safer_settings_table(:'test_stage'::text);
