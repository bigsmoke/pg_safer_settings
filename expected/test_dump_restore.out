-- createdb
-- psql -f 'sql/test_dump_restore.sql' -v 'extension_name=pg_safer_settings' -v 'test_stage=pre-dump'
psql:sql/test_dump_restore.sql:21: NOTICE:  installing required extension "pg_utility_trigger_functions"
psql:sql/test_dump_restore.sql:21: NOTICE:  installing required extension "hstore"
CREATE EXTENSION
CALL
-- pg_dump --format=custom --file <dump_file>
-- psql -f 'sql/test_dump_restore.sql' -v 'extension_name=pg_safer_settings' -v 'test_stage=post-restore'
CALL
